code/game/atoms_health.dm

```diff
@@ -0,0 +1,293 @@
+ * Returns `null` if health is not in use.
+ */
+/atom/proc/is_alive()
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	return get_current_health() > 0
+
+/**
+ * Health modification for the health system. Applies `health_mod` directly to `simple_health` via addition and calls `handle_death_change` as needed.
+ * Has no pre-modification checks, you should be using `damage_health()` or `restore_health()` instead of this.
+ * Returns `TRUE` if the death state changes, `null` if the atom is not using health, `FALSE` otherwise.
+ */
+/atom/proc/mod_health(health_mod, damage_type)
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	var/death_state = is_alive()
+	health_current = Clamp(health_current + health_mod, 0, get_max_health())
+	post_health_change(health_mod, damage_type)
+	var/new_death_state = is_alive()
+	if (death_state == new_death_state)
+		return FALSE
+	handle_death_change(new_death_state)
+	return TRUE
+
+/**
+ * Sets `health_current` to the new value, clamped between `0` and `health_max`.
+ * Has no pre-modification checks.
+ * Returns `TRUE` if the death state changes, `null` if the atom is not using health, `FALSE` otherwise.
+ */
+/atom/proc/set_health(new_health)
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	var/health_mod = new_health - health_current
+	return mod_health(health_mod)
+
+/**
+ * Restore's the atom's health by the given value. Returns `TRUE` if the restoration resulted in a death state change.
+ */
+/atom/proc/restore_health(damage, damage_type = null)
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	return mod_health(damage, damage_type = null)
+
+/**
+ * Damage's the atom's health by the given value. Returns `TRUE` if the damage resulted in a death state change.
+ * Resistance and weakness modifiers are applied here.
+ */
+/atom/proc/damage_health(damage, damage_type = null)
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+
+	// Apply resistance/weakness modifiers
+	damage *= get_damage_resistance(damage_type)
+
+	return mod_health(-damage, damage_type = null)
+
+/**
+ * Proc called after any health changes made by the system
+ */
+/atom/proc/post_health_change(health_mod, damage_type = null)
+	return
+
+/**
+ * Forces the atom to a state considered alive or dead. Should be overridden to make any necessary changes to ensure the atom's state is correct.
+ * Does not allow calling parent to avoid proc chain call ordering issues with `handle_death_change()` - Overrides should include their own calls to this proc once all necessary changes are complete.
+ */
+/atom/proc/set_death_state(new_death_state)
+	SHOULD_CALL_PARENT(FALSE)
+	if (!has_health())
+		return
+	if (new_death_state != is_alive())
+		return FALSE
+	health_current = new_death_state ? 0 : get_max_health()
+	handle_death_change(new_death_state)
+	return TRUE
+
+/**
+ * Ease of use proc for `set_death_state()`. This one kills the atom.
+ */
+/atom/proc/kill_health()
```

Comment:
Not called anywhere.

Reply:
Same as above. At this point I should probably port over the implementations to walls and items now rather than waiting for the base stuff to get reviewed.